LethalAPI.Core/Loader/Configs/ConfigLoader.cs

```diff
@@ -0,0 +1,263 @@
+    /// </summary>
+    public static void LoadAllConfigs()
+    {
+        Log.Debug($"deps loaded: {PluginLoader.Dependencies.Count}");
+
+        try
+        {
+            if (!LoadSeperatedConfigFiles)
+            {
+                LoadAllConfigsCombined();
+                return;
+            }
+
+            foreach (IPlugin<IConfig> plugin in PluginLoader.Plugins.Values)
+            {
+                LoadConfig(plugin);
+            }
+        }
+        catch (TypeLoadException e)
+        {
+            if (EmbeddedResourceLoader.Debug)
+            {
+                Log.Exception(e);
+            }
+        }
+        catch (Exception e)
+        {
+            Log.Error($"Could not load configs due to an error!", "LethalAPI-Loader");
+            Log.Debug($"Exception: \n{e}", EmbeddedResourceLoader.Debug, "LethalAPI-Loader");
+        }
+    }
+
+    /// <summary>
+    /// Loads the config for a specific plugin instance.
+    /// </summary>
+    /// <param name="plugin">The plugin instance to load the config for.</param>
+    public static void LoadConfig(IPlugin<IConfig> plugin)
+    {
+        if (LoadSeperatedConfigFiles)
+            plugin.UpdateConfig(GetSeparatedConfigValue(plugin));
+        else
+            plugin.UpdateConfig(GetConfigValueForPlugin(plugin));
+    }
+
+    [Pure]
+    private static Dictionary<string, object> GetLatestCombinedConfig()
+    {
+        string path = Path.Combine(ConfigDirectory, "lethal-api-config.yml");
+        Dictionary<string, object>? pluginConfigs = null;
+        try
+        {
+            if(File.Exists(path))
+                pluginConfigs = Serialization.Deserializer.Deserialize<Dictionary<string, object>>(File.ReadAllText(path));
+        }
+        catch (Exception e)
+        {
+            Log.Warn("Old config was invalid. A backup of the old config will be made.");
+            Log.Exception(e);
+            MakeBackupOfConfig(path);
+        }
+
+        pluginConfigs ??= new();
+
+        bool changed = false;
+        foreach (IPlugin<IConfig> plugin in PluginLoader.Plugins.Values)
+        {
+            if (!pluginConfigs.ContainsKey(plugin.Name.ToSnakeCase()))
+            {
+                Log.Info($"[Combined] Plugin config for plugin '{plugin.Name}' is missing! Generating new config.");
+                object? conf = plugin.Config;
+                try
+                {
+                    // ReSharper disable once NullCoalescingConditionIsAlwaysNotNullAccordingToAPIContract
+                    conf ??= Activator.CreateInstance(plugin.Config.GetType());
+                }
+                catch (Exception)
+                {
+                    Log.Error($"Could not find the default constructor for a config of type {plugin.Config.GetType()}.");
+                }
+
+                pluginConfigs.Add(plugin.Name.ToSnakeCase(), conf);
+                changed = true;
+            }
+
+            // object conf = pluginConfigs[plugin.Name.ToSnakeCase()];
```

Comment:
Can we remove these comments?

Reply:
I think we need to just take another look at this section in the future. This feature has to be toggled on, but it can make it easier for people to manage configs as they only have to deal with a single config file. 

As it stands right now, it only partially works. Immediately this isn't a problem, but if we do intend to allow it (which I think we should in the future) it will need a bit of overhaul. I left it in as a "legacy" to be built off of, but it should really just be rewritten.

Reply:
I fixed this to a working extent in my latest release. As for todo I try to fix things and remove the local todos before pring